lib/patme/implementation_builder.rb

```diff
@@ -11,47 +11,45 @@ def initialize(method_obj)
     end
 
     def build
-      values = get_values(method_header, method_params)
-      args = build_args(method_params, values)
-
-      Patme::Implementation.new(@method_obj, args)
+      Patme::Implementation.new(@method_obj, build_args)
     end
 
     private
 
-    def get_values(header, params)
-      return {} if params.size == 0
+    def get_values
+      return {} if method_params.empty?
+      return @_values if @_values
```

Comment:
I think there s no need for @_values variable since this method is being called only once. Can you please remove this line and `@_values = ` from line 32?